File: backend/models/route.py
from pydantic import BaseModel
from typing import Optional


class RouteStep(BaseModel):
    step: int
    instruction: str


class EvacuationRoute(BaseModel):
    id: str
    incidentId: str
    hotelId: str
    floor: int
    path: list[RouteStep]
    exitUsed: str
    isBlocked: bool = False
    estimatedTimeSeconds: int
    warning: Optional[str] = None
    updatedAt: str

File: frontend/types/route.ts
export interface RouteStep {
  step: number;
  instruction: string;
  x?: number;
  y?: number;
}

export interface EvacuationRoute {
  id: string;
  incidentId: string;
  hotelId: string;
  floor: number;
  path: RouteStep[];
  spatialData?: {
    guestPos: { x: number; y: number };
    exitPos?: { x: number; y: number } | null;
    dangerPos?: { x: number; y: number } | null;
  };
  exitUsed: string;
  isBlocked: boolean;
  estimatedTimeSeconds: number;
  warning?: string;
  updatedAt: string;
}
